src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php

```diff
@@ -0,0 +1,350 @@
+
+    /**
+     * @var CreateSimpleProductPageInterface
+     */
+    private $createSimpleProductPage;
+
+    /**
+     * @var CreateConfigurableProductPageInterface
+     */
+    private $createConfigurableProductPage;
+
+    /**
+     * @var IndexPageInterface
+     */
+    private $indexPage;
+
+    /**
+     * @var UpdateSimpleProductPageInterface
+     */
+    private $updateSimpleProductPage;
+
+    /**
+     * @var UpdateConfigurableProductPageInterface
+     */
+    private $updateConfigurableProductPage;
+
+    /**
+     * @var CurrentProductPageResolverInterface
+     */
+    private $currentPageResolver;
+
+    /**
+     * @var NotificationCheckerInterface
+     */
+    private $notificationChecker;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param CreateSimpleProductPageInterface $createSimpleProductPage
+     * @param CreateConfigurableProductPageInterface $createConfigurableProductPage
+     * @param IndexPageInterface $indexPage
+     * @param UpdateSimpleProductPageInterface $updateSimpleProductPage
+     * @param UpdateConfigurableProductPageInterface $updateConfigurableProductPage
+     * @param CurrentProductPageResolverInterface $currentPageResolver
+     * @param NotificationCheckerInterface $notificationChecker
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        CreateSimpleProductPageInterface $createSimpleProductPage,
+        CreateConfigurableProductPageInterface $createConfigurableProductPage,
+        IndexPageInterface $indexPage,
+        UpdateSimpleProductPageInterface $updateSimpleProductPage,
+        UpdateConfigurableProductPageInterface $updateConfigurableProductPage,
+        CurrentProductPageResolverInterface $currentPageResolver,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->createSimpleProductPage = $createSimpleProductPage;
+        $this->createConfigurableProductPage = $createConfigurableProductPage;
+        $this->indexPage = $indexPage;
+        $this->updateSimpleProductPage = $updateSimpleProductPage;
+        $this->updateConfigurableProductPage = $updateConfigurableProductPage;
+        $this->currentPageResolver = $currentPageResolver;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @Given I want to create a new simple product
+     */
+    public function iWantToCreateANewSimpleProduct()
+    {
+        $this->createSimpleProductPage->open();
+    }
+
+    /**
+     * @Given /^I want to create a new configurable product$/
+     */
+    public function iWantToCreateANewConfigurableProduct()
+    {
+        $this->createConfigurableProductPage->open();
+    }
+
+    /**
+     * @When I specify its code as :code
+     * @When I do not specify its code
+     */
+    public function iSpecifyItsCodeAs($code = null)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm([
```

Comment:
But the solution is not as trivial as it seems to be. Each page has different methods. Maybe it would be a good solution to assert proper interface after matching page?